source/manual/kubernetes-infrastructure.html.md

```diff
@@ -7,93 +7,175 @@ parent: "/manual.html"
+account-api-568d46d6f7-txwsx                                      2/2     Running            0                  34m
+account-api-dbmigrate-6n5mp                                       0/1     Completed          0                  35m
+account-api-worker-54597b666c-jlcfn                               1/1     Running            0                  34m
+asset-manager-7bd745d655-tzlgz                                    2/2     Running            0                  15h
+...
+```
 
-- To open a rails console:
-    - `k -n apps exec -it deploy/router-api -- rails c`
-- To open a shell:
-    - `k -n apps exec -it deploy/government-frontend -- bash`
-      \>> `curl` # (for example)
-- To open a shell on Router:
-    - `k -n apps exec -it deploy/router -c nginx`
+As you can see above, each pod either runs the application (we can see two pods running Account API, for example) or runs a worker or cron job for the application (we can see one pod running the Account API application worker).
 
-Note that you can omit the `-n apps` bit of the above commands if you specify a default namespace (recommended), by:
+You can fetch the logs for all running containers in a pod. Take one of the pods from the output above:
 
-- `k config set-context --current --namespace=apps`
+```sh
+k logs account-api-568d46d6f7-4pnld
+```
 
-#### Testing changes to a rake task.
+Or you can fetch the logs using its 'app deployment' name. This will choose one of the Account API pods at random, much like the old jumpbox methods:
 
-For special router publisher as an example:
+```sh
+k logs deploy/account-api
+```
 
-1. Deploy your branch (see above)
-2. Run the following command(s):
+## Open a rails console
+
+Open a rails console with:
 
+```sh
+k exec -it deploy/whitehall-admin -- rails c
 ```
-BASE_PATH=whatever
 
-kubectl run -n apps --image 172025368201.dkr.ecr.eu-west-1.amazonaws.com/special-route-publisher special-route-publisher -- rake "publish_one_special_route[${BASE_PATH?}]"
+If you try doing this having assumed a `readonly` role when setting context, you'll see a permissions error:
 
-kubectl logs -fnapps special-route-publisher
+```sh
+Defaulted container "app" out of: app, nginx
+Error from server (Forbidden): pods "whitehall-admin-6bb47c48d-wkkxk" is forbidden: User "christopher.ashton-user" cannot create resource "pods/exec" in API group "" in the namespace "apps"
+```
 
-kubectl -napps delete po/special-route-publisher
+You need write permissions to be able to open a console (`poweruser` or above).
+
+## Open a shell
+
+Open a shell (requires `poweruser` or above):
 
+```sh
+k exec -it deploy/whitehall-admin -- bash
 ```
 
-There’s also a [command in the PR](https://github.com/alphagov/special-route-publisher/pull/257) that might work.
+Note that the prompt will have prefix that may feel a little unfamiliar: `I have no name!@whitehall-admin-6bb47c48d-wkkxk:/app$`.
```

Comment:
This only happens for asset-manager and whitehall-admin, because (for tedious reasons to do with how NFS was set up on the old system) those have to run as a specific user ID that matches what's on NFS.

Reply:
Ooh interesting!